components/Form.svelte

```diff
@@ -50,9 +66,39 @@
   <br />
 {/if}
 
-<div>
+<div style="display: flex; justify-content: space-between">
   <button type="button" on:click={remove}>Delete</button>
-  <button type="button" on:click={clearCurrentlyEditing} style="float: right;"
-    >Save</button
+  <button type="button" on:click={() => (openModalForm = true)}
+    >Edit details</button
   >
+  <button type="button" on:click={clearCurrentlyEditing}>Save</button>
 </div>
+
+<Modal
+  passiveModal
+  hasScrollingContent
+  modalHeading={`Edit ${intervention_type} details`}
+  bind:open={openModalForm}
+>
+  {#each Object.entries(schema.properties) as [key, value]}
+    {#if value.enum}
+      <Dropdown
+        titleText={value.description || value.title}
+        items={value.enum.map((v) => ({ id: v, text: v }))}
+        bind:selectedId={details[key]}
+      />
+      <br />
+    {/if}
+  {/each}
+
+  {#if intervention_type == "route"}
+    <NumberInput
+      step={0.1}
+      label={routeSchema.properties.width.description}
+      bind:value={details.width}
+    />
+  {/if}
+
+  <!-- TODO Hack for the Modal -->
```

Comment:
The Carbon Modal is not smart about dropdowns that may expand, or has any way I can find to specify a height without something like this